CMakeLists.txt

```diff
@@ -52,30 +52,45 @@ message(STATUS "ROCMLIR_BIN_DIR: ${ROCMLIR_BIN_DIR}")
 set(CMAKE_BUILD_RPATH "${ROCMLIR_LIB_DIR};${LLVM_EXTERNAL_LIB_DIR}")
 message(STATUS "CMAKE_BUILD_RPATH: ${CMAKE_BUILD_RPATH}")
 
+set(CMAKE_INSTALL_PREFIX "/opt/rocm" CACHE STRING "")
+
+function(_maybe_force_set var val type docs force)
+  if (${force})
+    set(${var} ${val} CACHE ${type} "${docs}" FORCE)
+  else()
+    set(${var} ${val} CACHE ${type} "${docs}")
+  endif()
+endfunction()
+
+function(_set_librockcompiler_varying_bools value force)
+  _maybe_force_set(BUILD_SHARED_LIBS ${value} BOOL "" ${force})
+  _maybe_force_set(LLVM_BUILD_LLVM_DYLIB ${value} BOOL "" ${force})
+  _maybe_force_set(LLVM_BUILD_EXAMPLES ${value} BOOL "" ${force})
+  _maybe_force_set(MLIR_ENABLE_ROCM_RUNNER ${value} BOOL "" ${force})
+  _maybe_force_set(MLIR_INCLUDE_INTEGRATION_TESTS ${value} BOOL "" ${force})
+  _maybe_force_set(ROCMLIR_DRIVER_PR_E2E_TEST_ENABLED ${value}
+    BOOL "Enable build PR-triggered E2E tests for Rock driver" ${force})
+endfunction()
+
+# Set up defaults for not using librockcompiler either way
+_set_librockcompiler_varying_bools(ON OFF)
+
 if(BUILD_MIXR_TARGET)
   set(BUILD_FAT_LIBROCKCOMPILER ON CACHE BOOL "Build static librockCompiler")
   set(EXPORT_ALL_HEADERS ON CACHE BOOL "Export internal headers")
 endif()
 
 # Library type and linkage settings
-if( NOT DEFINED BUILD_FAT_LIBROCKCOMPILER )
+if(NOT DEFINED BUILD_FAT_LIBROCKCOMPILER)
   set(BUILD_FAT_LIBROCKCOMPILER OFF CACHE BOOL "Build fat librockCompiler to link into Rock driver")
```

Comment:
How about setting it FORCEly if it is not defined so that in Jenkinksfile, you do not have to explicitly define it as OFF.

Reply:
FORCE overrides settings from the command line

Reply:
If it is set from the command line we won't get in the `if`, right?

Reply:
No, we can, if you set the variable in a previous CMake run